### ISSUE TITLE
Query<UnmappedType>

### ISSUE THREAD

**Issue by radiy** [nhibernate/nhibernate-core#2167]
### Linq on unmapped type is NOP.

Consider code below it will executes successfully and do nothing.
It is clear that it is mistake on  programmer`s part.
Such operation should throw exception same as HQL.
```
    class Program
    {
        public class UnmappedClass
        {
        }
        static void Main(string[] args)
        {
            var configuration = new NHibernate.Cfg.Configuration();
            configuration.AddProperties(new Dictionary<string, string> {
                {NHibernate.Cfg.Environment.Dialect, typeof(MySQL5Dialect).FullName},
                {NHibernate.Cfg.Environment.ConnectionDriver, typeof(MySqlDataDriver).FullName},
                {NHibernate.Cfg.Environment.ConnectionString, "host=localhost; user=root"},
            });
            ISession session = configuration.BuildSessionFactory().OpenSession();
            session.Query<UnmappedClass>().Delete();
        }
    }
```

**Comment by fredericDelaporte** [comment]
May you describe explicitly (edit your first message) the current behavior: what happens when trying to delete an unmapped class?
Additionally, if you could check how HQL DML delete behaves with an unmapped class, it would be helpful too.

**Comment by radiy** [comment]
@fredericDelaporte I wrote small sample, to illustrate problem. If you compile and run, it will complete successfully. I want that in this case exception was thrown, because it is not first time when I trip over this behavior.
I undestand that nhiberrnate offers different ways for doing operations but I find this behavior contrintuitive and want to do something about it.

```

    class Program
    {
        public class UnmappedClass
        {
        }
        static void Main(string[] args)
        {
            var configuration = new NHibernate.Cfg.Configuration();
            configuration.AddProperties(new Dictionary<string, string> {
                {NHibernate.Cfg.Environment.Dialect, typeof(MySQL5Dialect).FullName},
                {NHibernate.Cfg.Environment.ConnectionDriver, typeof(MySqlDataDriver).FullName},
                {NHibernate.Cfg.Environment.ConnectionString, "host=localhost; user=root"},
            });
            ISession session = configuration.BuildSessionFactory().OpenSession();
            session.Query<UnmappedClass>().Delete();
        }
    }
```

**Comment by radiy** [comment]
> Additionally, if you could check how HQL DML delete behaves with an unmapped class, it would be helpful too.

HQL throws on unmapped class.
```
Unhandled Exception: NHibernate.Hql.Ast.ANTLR.QuerySyntaxException: UnmappedClass is not mapped [from UnmappedClass]
```

**Comment by fredericDelaporte** [comment]
Nowhere in your report you explicitly states what is the troubles. That is what I ask you to do. It seems the trouble is that the Linq query to delete an unmapped entity "executes" without throwing an exception, as if there was no trouble. Can you confirm this and make it clear in your first comment please?

Thanks for having check the HQL case.

(By the way , since HQL behaves as it should, there is no reason to believe there is any kind of "legacy" for not throwing when trying to delete an unmapped entity with Linq. DML with Linq is relatively new in NHibernate indeed, and not throwing on unmapped entity is just a bug. I think there is no need for a configuration option there: it should just throw when an invalid operation is attempted.)

**Comment by radiy** [comment]
I cleaned up first message as you suggested, please let me known if it is ok.
My assumption about "legacy" behavior is based on original Linq behavior session.Query<UnmappedClass>().ToList() do nothing and returns empty list.

So it is fine if I make merge request with exception on unmapped class?

**Comment by fredericDelaporte** [comment]
Yes, go ahead with a PR.

**Comment by mithrendal** [comment]
Any progress on this ? Just had the same problem on a simple query and searched for hours because there was no exception thrown by NHibernate. In the end it discoverd by a lengthy analysis that I forgot to map the class. After mapping the class everthing was fine... Would be very cool if this gets resolved in the future.

**Comment by radiy** [comment]
@mithrendal Hi, sorry, didn't make any progress on that one.

**Comment by mithrendal** [comment]
don't worry😀 this error happens to me only once all four years or so ... So by any luck when it happens again on me it is by a little chance already been fixed 🤓

**Comment by cyberkito** [comment]
It is problem when sesion.Query<UnmappedType>().ToList() or ..FirstOrDefault() does not throw exception. Real result for session.Query<MentToBeMapped>().ToList() should not be empty, but exception, so that problem with hbm mapping file can be fixed. 
I don't find any reason for this Query to return results without entity being mapped to any table. This is very problematic when MappingException only comes when mapping MentToBeMapped.hbm.xml is joined to other class in OtherClass.hbm.xml mapping file. Running the query should also notify problems in mapping, I think, so that problems can be noticed also without links to other mappings.

**Comment by bahusoid** [comment]
Duplicate of #1106

**Comment by fredericDelaporte** [comment]
I am not agree with setting it as a duplicate. It is very likely related, with the same root cause, but it has to be checked.
The case here is about DML Linq (a delete in this case), while #1106 is about regular querying Linq (a select).

**Comment by bahusoid** [comment]
I just added info about DML to #1106. And I checked sometime ago - possible fix throw inside `QueryExpressionPlan` if no translators provided should fix both issues;
